### PR TITLE
Refs [TASK][35609] Check Internet

### PR DESCRIPTION
**Pull request by zGararz**
## Related Tickets
- [#Task 35609: Check Internet](https://edu-redmine.sun-asterisk.vn/issues/35609)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.net.ConnectivityManager
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.example.itbook.R
+import com.example.itbook.utils.showMessage
+
+@Suppress("DEPRECATION")
```
Comment (datdt-1935) [inline app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt:12]: cái gì deprecated đây em?

File: app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    private fun isOnline(context: Context) = try {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val networkInfo = connectivityManager.activeNetworkInfo
+
+        (networkInfo != null && networkInfo.isConnected)
```
Comment (datdt-1935) [inline app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt:39]: dòng này có tác dụng gì?

File: app/src/main/java/com/example/itbook/ui/home/HomeFragment.kt [hunk 3]
```diff
@@ -68,26 +68,36 @@ class HomeFragment : BaseFragment(), HomeContract.View {
...
+        )
     }
 
     private fun onItemBookClick(currentCategory: Int, position: Int) {
         addFragment(
             DetailBookFragment.getInstance(
                 previewCategories[currentCategory].books[position].isbn13,
                 false
-            )
+            ), R.anim.anim_push_in, R.anim.anim_push_out
```
Comment (datdt-1935) [inline app/src/main/java/com/example/itbook/ui/home/HomeFragment.kt]: break line

File: app/src/main/res/anim/anim_push_in.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<translate xmlns:android="http://schemas.android.com/apk/res/android"
+    android:duration="300"
+    android:fromXDelta="100%"
+    android:toXDelta="0%"></translate>
```
Comment (datdt-1935) [inline app/src/main/res/anim/anim_push_in.xml]:
```suggestion
    android:toXDelta="0%"/>
```

File: app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt (hunk 1 above)
Comment (zGararz) [reply app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt:12]: cái activeNetworkInfo anh ạ

File: app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt (hunk 2 above)
Comment (zGararz) [reply app/src/main/java/com/example/itbook/broadcast/InternetReceiver.kt:39]: dòng này trả về true nếu có kết nối, false nếu không ạ

File: app/src/main/java/com/example/itbook/ui/main/MainActivity.kt [hunk 5]
```diff
@@ -16,7 +19,10 @@ class MainActivity() : BaseActivity() {
 
     private val favoriteFragment = FavoriteBooksFragment()
     private val homeFragment = HomeFragment()
+    private var receiver: BroadcastReceiver? = null
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/itbook/ui/main/MainActivity.kt]:
```suggestion
    private var receiver: InternetReceiver? = null
```
để thế này đc k

File: app/src/main/java/com/example/itbook/ui/main/MainActivity.kt [hunk 6]
```diff
@@ -38,6 +44,21 @@ class MainActivity() : BaseActivity() {
...
+    private fun registerInternetReceiver() {
+        val filter = IntentFilter("android.net.conn.CONNECTIVITY_CHANGE")
+        receiver = InternetReceiver()
+        registerReceiver(receiver, filter)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        try {
+            unregisterReceiver(receiver)
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/itbook/ui/main/MainActivity.kt]: check xem khác Null thì mới unregíster thôi

**Pull request merged**